File: src/backend/HiddenWing.BrandCenter.Domain/Entities/Theme.cs
namespace HiddenWing.BrandCenter.Domain.Entities;

/// <summary>
/// Reusable color and typography palette that products can inherit.
/// </summary>
public sealed class Theme : Entity
{
    public string Name { get; set; } = string.Empty;

    public string Slug { get; set; } = string.Empty;

    public string PrimaryColor { get; set; } = string.Empty;

    public string SecondaryColor { get; set; } = string.Empty;

    public string AccentColor { get; set; } = string.Empty;

    public string BackgroundColor { get; set; } = string.Empty;

    public string SurfaceColor { get; set; } = string.Empty;

    public string TextColor { get; set; } = string.Empty;

    public string MutedTextColor { get; set; } = string.Empty;

    public string BorderColor { get; set; } = string.Empty;

    public string FontFamily { get; set; } = string.Empty;

    public string BorderRadius { get; set; } = "8px";

    public bool IsDefault { get; set; }

    public ICollection<ProductBranding> ProductBrandings { get; set; } = new List<ProductBranding>();
}

File: src/frontend/hidden-wing-brand-center-web/src/types/theme.ts
export interface Theme {
  id: string;
  name: string;
  slug: string;
  primaryColor: string;
  secondaryColor: string;
  accentColor: string;
  backgroundColor: string;
  surfaceColor: string;
  textColor: string;
  mutedTextColor: string;
  borderColor: string;
  fontFamily: string;
  borderRadius: string;
  isDefault: boolean;
  createdAt: string;
  updatedAt: string;
}

export type ThemeWriteModel = Omit<Theme, "id" | "createdAt" | "updatedAt">;

export const emptyTheme = (): ThemeWriteModel => ({
  name: "",
  slug: "",
  primaryColor: "#1C3353",
  secondaryColor: "#3E536B",
  accentColor: "#C6A15B",
  backgroundColor: "#F4EFE6",
  surfaceColor: "#FFFFFF",
  textColor: "#1A1F29",
  mutedTextColor: "#5C6B7A",
  borderColor: "#D9D2C5",
  fontFamily: "Inter, system-ui, sans-serif",
  borderRadius: "8px",
  isDefault: false
});
